src/main/java/org/rumbledb/items/parsing/ItemParser.java

```diff
@@ -202,6 +210,18 @@ public static Item getItemFromObject(JsonReader<Object> object, ExceptionMetadat
         }
     }
 
+    private static void assertSingleItemAtTopLevel(JsonReader<Object> object, boolean isTopLevel)
+            throws java.io.IOException {
+        if (isTopLevel) {
+            while(object.getCurrentIndex() != object.length()) {
+                object.read();
+                if (object.last() != 32) {  // TODO: replace this with wasWhiteSpace() of DslJson::JsonReader
+                    throw new RuntimeException("More than one top level item detected in JSON");
```

Comment:
This should be a parsing exception. Actually, we should create a new exception JsonParsingException with error code FOJS0001 and use it everywhere else instead of ParsingException, because XPST0003 is normally only for query parsing exceptions and is thrown statically, while FOJS0001 is thrown dynamically.

(see https://www.w3.org/TR/xpath-functions-31/#func-parse-json )

Reply:
Also, the code for duplicate keys should be FOJS0003, caught by the ItemParser (not sure if this is what is thrown at the moment, as it is only caught later in the objectitem constructor I believe).

Reply:
Thanks for catching this as these should definitely be on our to-do list.

The RuntimeException is caught and converted to a ParsingException in the ItemParser at the moment. All ParsingExceptions thrown return XPST0003 I think.